packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -496,7 +496,7 @@ async function compileBicepToJson(
   bicepCommand: string,
   bicepOrchestrationFilePath: string
 ): Promise<JSON> {
-  const command = `${bicepCommand} build ${bicepOrchestrationFilePath} --stdout`;
+  const command = `${bicepCommand} build "${bicepOrchestrationFilePath}" --stdout`;
```

Comment:
is there any library to do the escape? I'm not sure whether `"` works for all cases

Reply:
+1

From my previous experience when implementing dependency checker, it is better to use `child_process.spawn` with `shell: false` and pass in an array to prevent shell escape issues. 
somethiong like: 
```
child_process.spawn([bicepCommand, "build", bicepOrchestrationFilePath, "--stdout"], {shell: false})
```

[Node.js docs](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options
)

Reply:
From [Node.js docs](https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback):
```typescript
const { exec } = require('child_process');

exec('"/path/to/test file/test.sh" arg1 arg2');
// Double quotes are used so that the space in the path is not interpreted as
// a delimiter of multiple arguments.
```

Reply:
what if the path contains `"`? the comment only said `"` solves the space issue in path. Since we need to escape the path, it's better to consider other corner cases if possible.
Of course if we confirm `"` works for other cases, we can use it.

Reply:
What do you mean by the path has `"`? `"` is not allowed to appeared in file or directory name.

Reply:
1. File name can contain `"` and [other special characters](https://ss64.com/osx/syntax-filenames.html) like `'` and `$` on macOS and Linux. 
2. If you use `child_process.exec`, Node.js will spawn a default shell (mostly `cmd.exe` on Windows and `/bin/bash` on macOS and Linux, but users can customize this which is not managed by us) and use the shell the execute the command. `"` can handle some cases. But some other examples of shell escape issues:
    - If there is a `$` in the path, bash will treat `$xxx` as an environment variable and replace them with actual values, which leads to incorrect result. 
    - There are even more dangerous cases, if the filename contains a `!something`, bash will interpret it as "executing the last command that starts with `something` in history".  If the last command is a `rm -rf xxx` or `shutdown`, this can cause issues. 
    - The users can set some other shell as their default shell. In this case we don't even know what the escape rules for these shells are.

    These are just examples of some shell escape issues and there are more.

Reply:
I will try to replace `child_process.exec()` with `child_process.spawn()` but will leave it to next PR to avoid making this PR to big.

[Task Item](https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/11043412)